onnxruntime/core/util/qmath.cc

```diff
@@ -30,16 +29,25 @@ void QGemmu8s8_s32(
 #ifdef MLAS_SUPPORTS_GEMM_U8X8
   MlasGemm(M, N, K, lhs_data, lda, lhs_offset, rhs_data, ldb, rhs_offset, result_data, ldc, thread_pool);
 #else
+  ORT_UNUSED_PARAMETER(M);
+  ORT_UNUSED_PARAMETER(N);
+  ORT_UNUSED_PARAMETER(K);
+  ORT_UNUSED_PARAMETER(lhs_data);
+  ORT_UNUSED_PARAMETER(lda);
+  ORT_UNUSED_PARAMETER(lhs_offset);
+  ORT_UNUSED_PARAMETER(rhs_data);
+  ORT_UNUSED_PARAMETER(ldb);
+  ORT_UNUSED_PARAMETER(rhs_offset);
+  ORT_UNUSED_PARAMETER(result_data);
+  ORT_UNUSED_PARAMETER(ldc);
   ORT_UNUSED_PARAMETER(thread_pool);
 
-  ORT_ENFORCE(lhs_offset == 0 && rhs_offset == 0, "For Eigen, zero point must be zero");
-  ORT_ENFORCE(lda == K && ldb == N && ldc == N, "For Eigen only RowMajor*RowMajor=RowMajor format is supported");
-
-  EigenCastGEMM<uint8_t, int8_t, int32_t>(lhs_data, rhs_data, result_data, M, N, K);
+  ORT_NOT_IMPLEMENTED("MatMulInteger: activation uint8 and weight int8 not supported on ARM");
```

Comment:
Why was this change to not implemented instead of just using the Eigen GEMM for zero points of zero? Doesn't the U8S8 matmul test still run here? I see there's code in the test to keep the zero points as zero.

Reply:
It's useless in practice to not support non-zero zero point for uint8. I would prefer to show a not-implemented message.